### ISSUE TITLE
Spark Report - Wrong test durations displayed in tests tab

### ISSUE THREAD

**Issue by grasshopper7** [extent-framework/extentreports-java#247]
The duration times are not handling time zones correctly in the tests section. They are getting appended. Seems the number_to_time function is converting duration to Date.
This happens for version 5.0.2 and 5.0.3. This works correctly in version 5.0.1. 

Modified the test.ftl to add the hour display to make it clear. `${test.timeTaken()?number_to_time?string("HH:mm:ss:SSS")}`

Display at UTC + 1.
![utc +1](https://user-images.githubusercontent.com/10829355/91142139-36698280-e6a8-11ea-8edd-aea206f1fa1c.png)

Display at UTC - 7.
![utc -7](https://user-images.githubusercontent.com/10829355/91142170-408b8100-e6a8-11ea-8980-15f855e583ce.png)

**Comment by grasshopper7** [comment]
public final String timeTaken() {
    	long duration = endTime.getTime() - startTime.getTime();
    	long millis = duration % 1000;
    	long second = (duration / 1000) % 60;
    	long minute = (duration / (1000 * 60)) % 60;
    	long hour = (duration / (1000 * 60 * 60));
    	
    	String durationText = String.format("%d:%d:%02d.%d", hour, minute, second, millis);
    	if (hour == 0 && minute == 0)
    		durationText = String.format("%02d.%d", second, millis);    		
    	else if (hour == 0)
    		durationText = String.format("%d:%02d.%d", minute, second, millis);    	
    	return durationText;
    }

Above seems to work by moving the duration calculation inside Test.java, though it includes hour portion also. Maybe there is a shorter solution.

**Comment by anshooarora** [comment]
@grasshopper7 The 1st solution (modifying the template) is more elegant since it represents the millis in the correct format.  Can you send a pull request?

**Comment by grasshopper7** [comment]
I think there is some confusion regarding the solution.

The `number_to_time` function in the template was added for issue [220](https://github.com/extent-framework/extentreports-java/issues/220). https://github.com/extent-framework/extentreports-java/blob/982a55d29c79ceb6c3d43cd4a62502392e5f7ca5/src/main/resources/com/aventstack/extentreports/templates/spark/partials/test.ftl#L62
I just manipulated the time pattern to add the hour to the time display `"HH:mm:ss:SSS"` so that the issue becomes clear. The `number_to_time` is not suitable for the purpose of displaying the duration in terms of mins, secs and millis. The function converts the long to a Java date.
The Java method in the comment above is the solution to generate the duration string. Though the hour portion can be removed.

Hope this is clearer.
